server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/oscfms/OscActionFsmBase.java

```diff
@@ -0,0 +1,378 @@
+                    flowInstanceID, taskStatus);
+        }
+        return ret;
+    }
+
+    /**
+     * start or resume FSM jump to create ghost table state for first start
+     * 
+     * @param schedulerID
+     * @param schedulerTaskID
+     */
+    public void start(Long schedulerID, Long schedulerTaskID) {
+        ScheduleEntity scheduleEntity = scheduleService.nullSafeGetById(schedulerID);
+        ScheduleTaskEntity scheduleTaskEntity = scheduleTaskService.nullSafeGetById(schedulerTaskID);
+        OnlineSchemaChangeScheduleTaskParameters parameters = JsonUtils.fromJson(scheduleTaskEntity.getParametersJson(),
+                OnlineSchemaChangeScheduleTaskParameters.class);
+        String currentState = parameters.getState();
+        // first yield, jump to create table state to decrease wait time
+        if (StringUtils.equals(currentState, OscStates.YIELD_CONTEXT.getState())) {
+            transferTaskStatesWithStates(OscStates.YIELD_CONTEXT.getState(), OscStates.CREATE_GHOST_TABLES.getState(),
+                    null, scheduleTaskEntity, TaskStatus.RUNNING);
+        } else {
+            // recover current state
+            scheduleTaskRepository.updateStatusById(schedulerTaskID, TaskStatus.RUNNING);
+        }
+        actionScheduler.submitFMSScheduler(scheduleEntity, schedulerTaskID);
+    }
+
+    public void cancel(Long schedulerID, Long schedulerTaskID) {
+        ScheduleEntity scheduleEntity = scheduleService.nullSafeGetById(schedulerID);
+        ScheduleTaskEntity scheduleTaskEntity = scheduleTaskService.nullSafeGetById(schedulerTaskID);
+        OnlineSchemaChangeScheduleTaskParameters parameters = JsonUtils.fromJson(scheduleTaskEntity.getParametersJson(),
+                OnlineSchemaChangeScheduleTaskParameters.class);
+        String currentState = parameters.getState();
+        transferTaskStatesWithStates(null, OscStates.CLEAN_RESOURCE.getState(),
+                "CANCEL", scheduleTaskEntity, TaskStatus.CANCELED);
+        // try submit scheduler in case task may in failed state
+        actionScheduler.submitFMSScheduler(scheduleEntity, schedulerTaskID);
+    }
+
+    /**
+     * check if task should continue
+     * 
+     * @param context
+     * @return
+     */
+    protected boolean isTaskExpired(OscActionContext context) {
+        ScheduleTaskEntity scheduleTask = context.getScheduleTask();
+        OnlineSchemaChangeScheduleTaskParameters parameters =
+                parseOnlineSchemaChangeScheduleTaskParameters(scheduleTask.getParametersJson());
+        // check task is expired
+        Long scheduleId = context.getSchedule().getId();
+        Long scheduleTaskId = scheduleTask.getId();
+        Duration between = Duration.between(scheduleTask.getCreateTime().toInstant(), Instant.now());
+        log.info("Schedule id {} to check schedule task status with schedule task id {}", scheduleId, scheduleTaskId);
```

Comment:
odc use key=value log style.

e.g. `log.info("Check schedule task status, scheduleId={}, scheduleTaskId={}", scheduleId, scheduleTaskId)`

key=value style is easy for log analyeis by machine.

so as below.

Reply:
it's a code migrate, not contains logic change. i will create another PR to fix this type issue